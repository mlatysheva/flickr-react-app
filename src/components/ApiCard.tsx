import { useContext } from 'react';
import { useNavigate } from 'react-router-dom';
import { ApiCardContext } from './ApiCardContext';

export interface ApiCardProps {
  id: string;
  dateupload: string;
  title: string;
  thumbnail: string;
  large_url: string;
  original_url: string;
  ownername?: string;
  tags?: string;
  views?: string;
}

export function ApiCard(props: ApiCardProps) {
  const navigate = useNavigate();
  const context = useContext(ApiCardContext);

  return (
    <div className="card small-api-card">
      <div className="title-wrapper">
        <img className="photo-image" src={props.thumbnail} alt="photo" />
        <p className="title">
          {props.title
            .split(' ')
            .slice(0, 6)
            .join(' ')
            .replace(/[,.\-!]\s*$/, ' ')}
        </p>
      </div>
      <button
        className="api-card-link"
        onClick={() => {
          context.card = props;
          navigate('/api/photo');
        }}
      >
        More...
      </button>
    </div>
  );
}

export default ApiCard;
