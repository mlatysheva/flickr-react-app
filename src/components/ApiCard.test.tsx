import { render, screen } from '@testing-library/react';
import { ApiCard } from './ApiCard';
import { MemoryRouter } from 'react-router-dom';

describe("ApiCard", () => {
  it("renders ApiCard component", () => {
    render(<MemoryRouter>
      <ApiCard id={'51992273600'} dateupload={'1649484030'} title={'8_DSC2660'} thumbnail={'https://live.staticflickr.com/65535/51992273600_0f8b1b4242_t.jpg'} large_url={'https://live.staticflickr.com/65535/51992273600_0f8b1b4242.jpg'} original_url={'https://live.staticflickr.com/65535/51992273600_0f8b1b4242_b.jpg'} />
    </MemoryRouter>);
    expect(screen.getAllByText(/8_DSC2660/i)[0]).toBeInTheDocument();
  })
})

test ("ApiPage as fragment matches the snapshot", () => {
  const { asFragment } = render(<MemoryRouter>
    <ApiCard id={'51956937593'} dateupload={"1648062917"} title={'Svínafellsjökull Glacier'} thumbnail={'https://live.staticflickr.com/65535/51956937593_4f1f3594b1_t.jpg'} large_url={'https://live.staticflickr.com/65535/51956937593_4f1f3594b1.jpg'} original_url={'https://live.staticflickr.com/65535/51956937593_4f1f3594b1_b.jpg'} />
  </MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
})

export {};
