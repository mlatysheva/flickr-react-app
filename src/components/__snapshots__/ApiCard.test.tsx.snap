// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiPage as fragment matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="card small-api-card"
  >
    <div
      class="title-wrapper"
    >
      <img
        alt="photo"
        class="photo-image"
        src="https://live.staticflickr.com/65535/51956937593_4f1f3594b1_t.jpg"
      />
      <p
        class="title"
      >
        Svínafellsjökull Glacier
      </p>
    </div>
    <button
      class="api-card-link"
    >
      More...
    </button>
  </div>
</DocumentFragment>
`;
